src/main/java/fruitshop/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,26 @@
+package fruitshop.strategy;
+
+import fruitshop.model.FruitTransaction;
+import fruitshop.model.Storage;
+
+public class PurchaseOperation implements OperationHandler {
+    public static final String READ_FILE_PATH = "src/main/resources/reportToRead.csv";
+    private final Storage storage;
+
+    public PurchaseOperation(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void handleTransaction(FruitTransaction fruitTransaction) {
+        int newQuantity = storage.getQuantity(fruitTransaction.getFruit())
+                - fruitTransaction.getQuantity();
+        if (newQuantity >= 0) {
+            storage.put(fruitTransaction.getFruit(), newQuantity);
+        } else {
+            throw new RuntimeException("negative balance " + newQuantity
+                    + " cannot be recorded at "
+                    + PurchaseOperation.class + " from fail " + READ_FILE_PATH);
```

Comment:
The exception message is misleading because it refers to a constant `READ_FILE_PATH` which suggests that the error is related to reading from a file, while the actual problem is the result of a negative balance after a purchase operation. The message should be clear and related to the actual issue. Instead, you can provide a message that includes the fruit and the attempted quantity that led to the negative balance.